hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieLogFile.java

```diff
@@ -38,49 +39,53 @@
   public static final String DELTA_EXTENSION = ".log";
   public static final Integer LOGFILE_BASE_VERSION = 1;
 
-  private final Path path;
-  private Optional<FileStatus> fileStatus;
+  private final transient Path path;
+  private final String pathStr;
+  private final String fileName;
+  // TODO: varadarb : Use Guava Optional everywhere except public API
+  private com.google.common.base.Optional<Long> fileLen;
 
   public HoodieLogFile(FileStatus fileStatus) {
     this(fileStatus.getPath());
-    this.fileStatus = Optional.of(fileStatus);
+    this.fileLen = com.google.common.base.Optional.of(fileStatus.getLen());
```

Comment:
Guava Optional is serializable. To avoid cascadiing changes, have limited the change from java.util.Optional to Guava Optional to only needed places